Providers/Modules/Plugins/Common/plugin/oms_configuration.rb

```diff
@@ -212,8 +238,24 @@ def load_configuration(conf_path, cert_path, key_path)
 
         File.open(conf_path).each_line do |line|
           if line =~ /AZURE_RESOURCE_ID/
-            @@AzureResourceId = line.sub("AZURE_RESOURCE_ID=","").strip
-            Thread.new(&method(:update_azure_resource_id)) if @@AzureResIDThreadLock.try_lock
+            # We have contract with AKS team about how to pass AKS specific resource id.
+            # As per contract, AKS team before starting the agent will set environment variable 
+            # 'custom-resourceId'
+            @@AzureResourceId = ENV['custom-resourceId']
```

Comment:
do we need to do any other sanity checks for the customer passed values other than checking for null?